lua/terrortown/entities/roles/spy/shared.lua

```diff
@@ -133,4 +172,56 @@ else
 			return false
 		end
 	end)
+
+	hook.Add("TTTCanSearchCorpse", "TTT2SpyChangeCorpseToTraitor", function(ply, corpse)	
+		if not GetConVar("ttt2_spy_confirm_as_traitor"):GetBool() then return end
+		
+		if corpse and corpse.was_role == ROLE_SPY and not corpse.reverted_spy then	
+			corpse.was_role = ROLE_TRAITOR
+			corpse.role_color = GetRoleByIndex(ROLE_TRAITOR).color
+			corpse.is_spy_corpse = true
+		end
+		
+	end)
+
+	hook.Add("TTT2ConfirmPlayer", "TTT2SpyChangeRoleToTraitor", function(confirmed, finder, corpse)	

```

Comment:
Now you've added TTT2SpecialRoleSyncing hook, is this hook still needed?

Reply:
Yes, since we don't want to call "SendFullStateUpdate" after every BodyFound hook, which would be required otherwise.